### DISCUSSION TITLE
New Line in payload

### DISCUSSION THREAD

**Discussion by kc9ryt** [emqx/emqx#12146] [Q&A] [answered]
Hello,

Im my example below i am publishing this message to a mqtt topic and then republishing it to an api. The \n characters are being interpreted as a new line when the republish is happening but that is breaking the JSON parse. Is there a way to just take the payload as in and not interpret the new line tag?

PAYLOAD SENT:
{
    "_id":"e739d848-20ab-42ff-8aa6-d9c96e3361d6",
    "text":"Two\nLines",
    "image":"",
    "createdAtt":"2023-12-11T23:16:21.469Z",
    "user":"redacted",
    "received":true
}

PAYLOAD REPUBLISHED:
{
    "sender" : "system-autoreply",
    "type" :  "ack",
    "msg" : "Message Received",
    "msgid" : "e739d848-20ab-42ff-8aa6-d9c96e3361",
    "recmesg" : "Two
Lines"
}

**Comment by keynslug** [comment]
Hi @kc9ryt, could you please share you rule / action configuration? This should help to understand where exactly in the pipeline it's happening?

**Comment by kc9ryt** [comment]
@keynslug The republish Action works file. The issue is with the HTTP Server action.

<img width="1324" alt="Screenshot 2023-12-13 at 8 20 37 AM" src="https://github.com/emqx/emqx/assets/3506952/efbfdaa6-d4b2-4b50-9839-6bf7a382729a">
<img width="1433" alt="Screenshot 2023-12-13 at 8 20 52 AM" src="https://github.com/emqx/emqx/assets/3506952/31757773-a141-4a21-87ed-f2e6eeedbd2a">
<img width="1429" alt="Screenshot 2023-12-13 at 8 21 14 AM" src="https://github.com/emqx/emqx/assets/3506952/e9ecce1e-9313-4930-bfb9-49a257d99c22">
<img width="1437" alt="Screenshot 2023-12-13 at 8 21 30 AM" src="https://github.com/emqx/emqx/assets/3506952/64a0ebee-af80-41a1-8539-ab1476498f40">

**Comment by keynslug** [comment, marked as answer]
I see. In the first instance, you could try rewriting your rule slightly to
```
SELECT
  payload,
  json_encode(payload.text) as json_text
FROM "PTPMSG/+/send"
```
...And then use it in the template directly, i.e. `"recmesg": ${json_text}` (note the missing quotes around the placeholder).

**Comment by kc9ryt** [comment]
That worked thanks so much for your help!
